go/trivial.go

```diff
@@ -3,33 +3,64 @@ package main
 import (
 	"fmt"
 	"os"
+	"strings"
 	"time"
 
-	"github.com/uber/jaeger-client-go"
+	jaeger "github.com/uber/jaeger-client-go"
 	"github.com/uber/jaeger-client-go/config"
 
-	"github.com/opentracing/opentracing-go"
+	opentracing "github.com/opentracing/opentracing-go"
 )
 
+func errExit(errstr string) {
+	fmt.Println(errstr)
+	os.Exit(1)
+}
+
 func main() {
-	// 1) Create a opentracing.Tracer that sends data to Zipkin
-	cfg := config.Configuration{
-		Sampler: &config.SamplerConfig{
-			Type:	"const",
-			Param:	1,
-		},
-		Reporter: &config.ReporterConfig{
-			LogSpans:		true,
-			BufferFlushInterval:	1 * time.Second,
-		},
+	var (
+		hostPort string
+		cfg      *config.Configuration
+	)
+
+	if len(os.Args) == 2 {
+		if os.Args[1] == "env" {
+			var err error
+			cfg, err = config.FromEnv()
+			if err != nil {
+				errExit(fmt.Sprintf("Cannot create configuration from environment: %+v\n", err))
+			}
+		} else {
+			hostPort = os.Args[1]
+			if !strings.Contains(os.Args[1], ":") {
+				hostPort += ":6831"
```

Comment:
what's the goal here? It silently ignores the input, not the best solution.

Also, I don't think this complication of the example is overall helpful. It might be better to just always use FromEnv, but if config.Sampler == nil then override it with const/1